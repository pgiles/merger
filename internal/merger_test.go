package internal

import (
	"bytes"
	"encoding/csv"
	"testing"

	approvals "github.com/approvals/go-approval-tests"
)

// Tests with E2E (End-to-End) prefix are not executed by "go test" for hopefully obvious reasons.

func init() {
	approvals.UseFolder("fixtures")
}

func TestCombine(t *testing.T) {
	m := new(Merger)
	files := []string{"../cmd/fixtures/test.csv", "../cmd/fixtures/transactions.CSV"}
	headers := []string{"first_name", "ssn", "Transaction Date", "Category", "Amount", "Amount", "ssn"}
	w := bytes.NewBufferString("")
	m.combine(csv.NewWriter(w), files, headers)
	//fmt.Print(w)

	approvals.VerifyString(t, w.String())
}
