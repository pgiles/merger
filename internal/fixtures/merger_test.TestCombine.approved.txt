first_name,ssn
John,123456
Kathy,687987
Bob,3979870
Transaction Date,Category,Amount
20221231,Merchandise,12.36
20230115,Grocery,68.77
20230131,Dining,39.98
